File: benchmarks/pandas/bench_dt_dayofyear_weekday.py
"""Benchmark: dt_dayofyear_weekday — pandas dt.dayofyear, dt.weekday on 100k values"""
import json
import time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = pd.date_range("2020-01-01", periods=ROWS, freq="D")
s = pd.Series(data)

for _ in range(WARMUP):
    _ = s.dt.dayofyear
    _ = s.dt.weekday

start = time.perf_counter()
for _ in range(ITERATIONS):
    _ = s.dt.dayofyear
    _ = s.dt.weekday
total = (time.perf_counter() - start) * 1000

print(json.dumps({"function": "dt_dayofyear_weekday", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))

File: benchmarks/tsb/bench_dt_dayofyear_weekday.ts
/**
 * Benchmark: dt_dayofyear_weekday — DatetimeAccessor dayofyear, weekday on 100k values
 */
import { Series } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const now = Date.now();
const data = Array.from({ length: ROWS }, (_, i) => new Date(now + i * 86_400_000));
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  s.dt.dayofyear();
  s.dt.weekday();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  s.dt.dayofyear();
  s.dt.weekday();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "dt_dayofyear_weekday",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
